src/query/block/types.go

```diff
@@ -183,3 +183,28 @@ func TakeLast(values ts.Datapoints) float64 {
 
 	return math.NaN()
 }
+
+// TimeTransform transforms a timestamp.
+type TimeTransform func(time.Time) time.Time
+
+// MetaTransform transforms meta data.
+type MetaTransform func(meta Metadata) Metadata
+
+// ValueTransform transform a float64.
+type ValueTransform func(float64) float64
+
+// LazyOpts describes options for lazy blocks.
+type LazyOpts interface {
```

Comment:
nit: All our options types have the full name, this should be `LazyOptions` instead of `LazyOpts` for consistency with the other option types names.